build.sh

```diff
@@ -346,6 +348,71 @@ function build_android {
     cd ../..
 }
 
+function ensure_ios_toolchain {
+    local TOOLCHAIN_PATH="build/toolchain-arm64-mac-ios.cmake"
+    if [ -e ${TOOLCHAIN_PATH} ]; then
+        echo "iOS arm64 toolchain file exists."
+        return 0
+    fi
+    echo
+    echo "iOS arm64 toolchain file does not exist."
+    echo "It will automatically be downloaded from http://opensource.apple.com."
+    read -p "Continue? (y/n) " -n 1 -r
+    echo
+    if [[ ! "$REPLY" =~ ^[Yy]$ ]]; then
+        echo "Toolchain file must be downloaded to continue."
+        exit 1
+    fi
+    curl -o "${TOOLCHAIN_PATH}" "${IOS_TOOLCHAIN_URL}" || {
+        echo "Error downloading iOS toolchain file."
+        exit 1
+    }
+    cat build/toolchain-arm64-mac-ios.filament.cmake >> ${TOOLCHAIN_PATH}
+    echo "Successfully downloaded iOS toolchain file and appended Filament-specific settings."
+}
+
+function build_ios_target {
+    local LC_TARGET=`echo $1 | tr '[:upper:]' '[:lower:]'`
+    local ARCH=$2
+
+    echo "Building iOS $LC_TARGET ($ARCH)..."
+    mkdir -p out/cmake-ios-${LC_TARGET}-${ARCH}
```

Comment:
Since there is one and only one arch with iOS, I suppose we could remove the arch suffix here.  On the other hand, this is more consistent with the Android build.

With WASM, we simply create `cmake-webgl-release` and `cmake-webgl-debug`.

Reply:
Looking ahead, I wonder if it would make more sense for the output suffix to be the backend API (vk/mtl/gl) rather than `arm64`.

Reply:
Yeah, I was following the Android conventions. I think having the architecture here is nice- if I can get OpenGL working, then _in theory_ we could also support the iOS Simulator, in which case the build directory would be `out/cmake-ios-debug-x86-64`. Not opposed to also adding the backend API to the build dir in the future.

Reply:
Ah, makes sense!